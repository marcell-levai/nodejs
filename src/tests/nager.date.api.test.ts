import axios from 'axios';

const customSerializer = {
  print(val: any): string {
    return JSON.stringify(val, null, 2);
  },
  test(val: any): boolean {
    return val && typeof val === 'object' && val.constructor.name === 'Object';
  },
};

expect.addSnapshotSerializer(customSerializer);

describe('Nager.Date API E2E tests', () => {
  const baseUrl = 'https://date.nager.at/api/v3';

  describe('NextPublicHolidays endpoint', () => {
    it('should retrieve the next public holidays for a country', async () => {
      const countryCode = 'US';

      const response = await axios.get(`${baseUrl}/NextPublicHolidays/${countryCode}`);

      expect(response.status).toBe(200);
      expect(response.data).toMatchSnapshot();
    });
  });

  describe('PublicHolidays endpoint', () => {
    it('should retrieve the public holidays for a specific year and country', async () => {
      const countryCode = 'US';
      const year = 2022; // Replace with the desired year

      const response = await axios.get(`${baseUrl}/PublicHolidays/${year}/${countryCode}`);

      expect(response.status).toBe(200);
      expect(response.data).toMatchSnapshot();
    });
  });
});