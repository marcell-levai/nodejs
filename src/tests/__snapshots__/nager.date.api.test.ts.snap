// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nager.Date API E2E tests NextPublicHolidays endpoint should retrieve the next public holidays for a country 1`] = `
[
  {
  "date": "2024-01-15",
  "localName": "Martin Luther King, Jr. Day",
  "name": "Martin Luther King, Jr. Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2024-02-19",
  "localName": "Presidents Day",
  "name": "Washington's Birthday",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2024-03-29",
  "localName": "Good Friday",
  "name": "Good Friday",
  "countryCode": "US",
  "fixed": false,
  "global": false,
  "counties": [
    "US-CT",
    "US-DE",
    "US-HI",
    "US-IN",
    "US-KY",
    "US-LA",
    "US-NC",
    "US-ND",
    "US-NJ",
    "US-TN"
  ],
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2024-05-27",
  "localName": "Memorial Day",
  "name": "Memorial Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2024-06-19",
  "localName": "Juneteenth",
  "name": "Juneteenth",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": 2021,
  "types": [
    "Public"
  ]
},
  {
  "date": "2024-07-04",
  "localName": "Independence Day",
  "name": "Independence Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2024-09-02",
  "localName": "Labor Day",
  "name": "Labour Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2024-10-14",
  "localName": "Columbus Day",
  "name": "Columbus Day",
  "countryCode": "US",
  "fixed": false,
  "global": false,
  "counties": [
    "US-AL",
    "US-AZ",
    "US-CO",
    "US-CT",
    "US-GA",
    "US-ID",
    "US-IL",
    "US-IN",
    "US-IA",
    "US-KS",
    "US-KY",
    "US-LA",
    "US-ME",
    "US-MD",
    "US-MA",
    "US-MS",
    "US-MO",
    "US-MT",
    "US-NE",
    "US-NH",
    "US-NJ",
    "US-NM",
    "US-NY",
    "US-NC",
    "US-OH",
    "US-OK",
    "US-PA",
    "US-RI",
    "US-SC",
    "US-TN",
    "US-UT",
    "US-VA",
    "US-WV"
  ],
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2024-11-11",
  "localName": "Veterans Day",
  "name": "Veterans Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2024-11-28",
  "localName": "Thanksgiving Day",
  "name": "Thanksgiving Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": 1863,
  "types": [
    "Public"
  ]
},
  {
  "date": "2024-12-25",
  "localName": "Christmas Day",
  "name": "Christmas Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2025-01-01",
  "localName": "New Year's Day",
  "name": "New Year's Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
]
`;

exports[`Nager.Date API E2E tests PublicHolidays endpoint should retrieve the public holidays for a specific year and country 1`] = `
[
  {
  "date": "2021-12-31",
  "localName": "New Year's Day",
  "name": "New Year's Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2022-01-17",
  "localName": "Martin Luther King, Jr. Day",
  "name": "Martin Luther King, Jr. Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2022-02-21",
  "localName": "Presidents Day",
  "name": "Washington's Birthday",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2022-04-15",
  "localName": "Good Friday",
  "name": "Good Friday",
  "countryCode": "US",
  "fixed": false,
  "global": false,
  "counties": [
    "US-CT",
    "US-DE",
    "US-HI",
    "US-IN",
    "US-KY",
    "US-LA",
    "US-NC",
    "US-ND",
    "US-NJ",
    "US-TN"
  ],
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2022-04-15",
  "localName": "Good Friday",
  "name": "Good Friday",
  "countryCode": "US",
  "fixed": false,
  "global": false,
  "counties": [
    "US-TX"
  ],
  "launchYear": null,
  "types": [
    "Optional"
  ]
},
  {
  "date": "2022-05-30",
  "localName": "Memorial Day",
  "name": "Memorial Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2022-06-20",
  "localName": "Juneteenth",
  "name": "Juneteenth",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": 2021,
  "types": [
    "Public"
  ]
},
  {
  "date": "2022-07-04",
  "localName": "Independence Day",
  "name": "Independence Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2022-09-05",
  "localName": "Labor Day",
  "name": "Labour Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2022-10-10",
  "localName": "Columbus Day",
  "name": "Columbus Day",
  "countryCode": "US",
  "fixed": false,
  "global": false,
  "counties": [
    "US-AL",
    "US-AZ",
    "US-CO",
    "US-CT",
    "US-GA",
    "US-ID",
    "US-IL",
    "US-IN",
    "US-IA",
    "US-KS",
    "US-KY",
    "US-LA",
    "US-ME",
    "US-MD",
    "US-MA",
    "US-MS",
    "US-MO",
    "US-MT",
    "US-NE",
    "US-NH",
    "US-NJ",
    "US-NM",
    "US-NY",
    "US-NC",
    "US-OH",
    "US-OK",
    "US-PA",
    "US-RI",
    "US-SC",
    "US-TN",
    "US-UT",
    "US-VA",
    "US-WV"
  ],
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2022-11-11",
  "localName": "Veterans Day",
  "name": "Veterans Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
  {
  "date": "2022-11-24",
  "localName": "Thanksgiving Day",
  "name": "Thanksgiving Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": 1863,
  "types": [
    "Public"
  ]
},
  {
  "date": "2022-12-26",
  "localName": "Christmas Day",
  "name": "Christmas Day",
  "countryCode": "US",
  "fixed": false,
  "global": true,
  "counties": null,
  "launchYear": null,
  "types": [
    "Public"
  ]
},
]
`;
